frontend/src/lib/functions/user.js

```diff
@@ -1,12 +1,23 @@
-import { accessLevel, startPage } from "../../stores";
+import { userName, userMail, accessLevel, startPage } from "../../stores";
 import { accessLevels } from "../types";
 
-export function getAccessLevel() {
-    return accessLevels.default
-}
+export const getUserData = async () => {
+    await fetch("/user", { method: "GET" }).then((response) => {
+      if (response.status == 200) {
+        response.json().then((data) => {
+          userName.set(data["1"].user_name);
```

Comment:
I don't know if `data["1"]` is always correct here. (not much JS knowledge) But when the `user_id` isn't == 1 then this could be `null`. The returned usr data could be (if `user_id=1`):
```JSON
{
    "1": {
        "user_id": 1,
        "user_name": "John Doe",
        "user_mail": "john.doe@example.com"
    }
}
```
or for example it could be (if `user_id=25`):
```JSON
{
    "25": {
        "user_id": 25,
        "user_name": "John Doe",
        "user_mail": "john.doe@example.com"
    }
}
```

Maybe you can access the element somehow else?

Reply:
I did not know that the "1" represents the user id. Maybe it would be easier to just send a list and not use the id as a key, because the id is in the dictionary anyways. This would make handling user data easier for the frontend.

Reply:
Î also understood it as some sort of index and not the user_id... quite confusing but if it is not possible to change it in the backend you could convert the values of this dict into an array and access the first element?

Reply:
@scrap111 It's already changed in the backend in the newest pull request (#102).
If you send an empty GET request to /user the backend now outputs the user_data of the logged in user like that:
```JSON
{
        "user_id": 25,
        "user_name": "John Doe",
        "user_mail": "1@example.com",
        "user_role": "User"
}
```
If you send a normal GET (with query params), then the result will be the same as before.

Reply:
My intention to this approach was to ease access to the returned elements. The key would always be the `user_id`. Yes, this is redundant because the `user_id` is also returned as a field of the value.

I already thought about changing the structure and return multiple elements as an array and not as a dictionary.